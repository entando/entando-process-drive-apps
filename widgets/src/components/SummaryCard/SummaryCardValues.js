import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import i18next from 'i18next';
import withStyles from '@material-ui/core/styles/withStyles';
import Typography from '@material-ui/core/Typography';
import Skeleton from '@material-ui/lab/Skeleton';
import { UpTrend as UpTrendIcon, DownTrend as DownTrendIcon } from 'components/common/Icons';

const styles = {
  root: {
    display: 'flex',
    padding: '20px 25px',
    justifyContent: 'space-between',
    alignItems: 'flex-end',
    '&.downTrend $trendValue': {
      color: '#ED5565',
    },
    '&.upTrend $trendValue': {
      color: '#23C6C8',
    },
  },
  totalLabel: {
    marginTop: 15,
  },
  trendValue: {
    fontWeight: 'bold',
  },
  valueIcon: {
    width: 13,
    height: 13,
  },
};

const SummaryCardValues = ({ values, classes, loading }) => {
  const generateSkeleton = () => (
    <>
      <div>
        <Skeleton width="110px" height={37} variant="rect" />
        <Skeleton width="90px" height={14} variant="rect" className={classes.totalLabel} />
      </div>
      <div>
        <Skeleton width="50px" height={16} variant="rect" />
      </div>
    </>
  );

  const trendNotation = trend => {
    switch (trend) {
      case 'up':
        return <UpTrendIcon className={classes.valueIcon} />;
      case 'down':
        return <DownTrendIcon className={classes.valueIcon} />;
      default:
        return null;
    }
  };

  const percVal = values && values.percentage ? Math.round(values.percentage) : null;
  const percent = !Number.isNaN(percVal) ? `${percVal < 0 ? 0 - percVal : percVal}%` : '-';

  const trend = percVal === 0 ? '' : percVal >= 0 ? 'up' : 'down';

  return (
    <div className={classNames(classes.root, `${trend}Trend`)}>
      {loading && generateSkeleton()}
      {!loading && values && (
        <>
          <div>
            <Typography variant="h4">{values.total}</Typography>
            <Typography variant="caption">
              {i18next.t(`card.labels.${values.totalLabel}`)}
            </Typography>
          </div>
          <div>
            <Typography variant="body2" component="span" className={classes.trendValue}>
              {percent} {trendNotation(trend)}
            </Typography>
          </div>
        </>
      )}
    </div>
  );
};

SummaryCardValues.propTypes = {
  classes: PropTypes.shape({
    root: PropTypes.string,
    totalLabel: PropTypes.string,
    trendValue: PropTypes.string,
    valueIcon: PropTypes.string,
  }).isRequired,
  loading: PropTypes.bool,
  values: PropTypes.shape({
    totalLabel: PropTypes.string,
    total: PropTypes.string,
    percentage: PropTypes.string,
  }),
};

SummaryCardValues.defaultProps = {
  loading: false,
  values: {},
};

export default withStyles(styles)(SummaryCardValues);
