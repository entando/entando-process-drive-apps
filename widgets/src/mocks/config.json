{
  "payload": {
    "code": "pda_task_list",
    "config": {
      "groups": "[{\"label\":\"Administrators\",\"key\":\"Administrators\",\"checked\":true},{\"label\":\"broker\",\"key\":\"broker\",\"checked\":true},{\"label\":\"admin\",\"key\":\"admin\",\"checked\":false},{\"label\":\"rest-all\",\"key\":\"rest-all\",\"checked\":false},{\"label\":\"kiemgmt\",\"key\":\"kiemgmt\",\"checked\":true},{\"label\":\"kie-server\",\"key\":\"kie-server\",\"checked\":true},{\"label\":\"appraiser\",\"key\":\"appraiser\",\"checked\":false},{\"label\":\"manager\",\"key\":\"manager\",\"checked\":true},{\"label\":\"supplier\",\"key\":\"supplier\",\"checked\":true},{\"label\":\"IT\",\"key\":\"IT\",\"checked\":true},{\"label\":\"PM\",\"key\":\"PM\",\"checked\":true}]",
      "options": "[{\"key\":\"newPageOnClick\",\"label\":\"Open new page on table row click\",\"checked\":false},{\"key\":\"showClaim\",\"label\":\"Show Claim Button\",\"checked\":true},{\"key\":\"showComplete\",\"label\":\"Show Complete Button\",\"checked\":true}]",
      "columns": "[{\"name\":\"task-is-skipable\",\"position\":0,\"isVisible\":false},{\"name\":\"task-parent-id\",\"position\":1,\"isVisible\":false},{\"name\":\"task-created-by\",\"position\":2,\"isVisible\":true},{\"name\":\"task-input-data.application.property.saleprice\",\"position\":3,\"isVisible\":true},{\"name\":\"task-input-data.TaskName\",\"position\":4,\"isVisible\":true},{\"name\":\"initiator\",\"position\":5,\"isVisible\":true},{\"name\":\"task-created-on\",\"position\":6,\"isVisible\":true},{\"name\":\"task-priority\",\"position\":7,\"isVisible\":false},{\"name\":\"task-proc-def-id\",\"position\":8,\"isVisible\":true},{\"name\":\"task-form\",\"position\":9,\"isVisible\":true},{\"name\":\"task-skippable\",\"position\":10,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.name\",\"position\":11,\"isVisible\":true},{\"name\":\"task-actual-owner\",\"position\":12,\"isVisible\":true},{\"name\":\"task-activation-time\",\"position\":13,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.address\",\"position\":14,\"isVisible\":true},{\"name\":\"task-input-data.application.errors\",\"position\":15,\"isVisible\":true},{\"name\":\"task-name\",\"position\":16,\"isVisible\":true},{\"name\":\"task-input-data.application.property.address\",\"position\":17,\"isVisible\":true},{\"name\":\"task-subject\",\"position\":18,\"isVisible\":true},{\"name\":\"containerId\",\"position\":19,\"isVisible\":true},{\"name\":\"task-container-id\",\"position\":20,\"isVisible\":true},{\"name\":\"processDefinitionId\",\"position\":21,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.ssn\",\"position\":22,\"isVisible\":true},{\"name\":\"task-description\",\"position\":23,\"isVisible\":true},{\"name\":\"task-input-data.Skippable\",\"position\":24,\"isVisible\":true},{\"name\":\"task-workitem-id\",\"position\":25,\"isVisible\":true},{\"name\":\"task-expiration-time\",\"position\":26,\"isVisible\":true},{\"name\":\"task-input-data.application.property.age\",\"position\":27,\"isVisible\":true},{\"name\":\"task-status\",\"position\":28,\"isVisible\":true},{\"name\":\"task-input-data.application.mortgageamount\",\"position\":29,\"isVisible\":true},{\"name\":\"task-process-instance-id\",\"position\":30,\"isVisible\":true},{\"name\":\"task-input-data.ActorId\",\"position\":31,\"isVisible\":true},{\"name\":\"task-process-id\",\"position\":32,\"isVisible\":true},{\"name\":\"task-input-data.application.property.locale\",\"position\":33,\"isVisible\":true},{\"name\":\"task-input-data.application.amortization\",\"position\":34,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.creditrating\",\"position\":35,\"isVisible\":true},{\"name\":\"application\",\"position\":36,\"isVisible\":true},{\"name\":\"task-input-data.NodeName\",\"position\":37,\"isVisible\":true},{\"name\":\"task-proc-inst-id\",\"position\":38,\"isVisible\":true},{\"name\":\"task-input-data.application.downpayment\",\"position\":39,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.annualincome\",\"position\":40,\"isVisible\":true},{\"name\":\"task-id\",\"position\":41,\"isVisible\":true}]",
      "knowledgeSource": "kieStaging",
      "process": "itorders.orderhardware@itorders_1.0.0-SNAPSHOT"
    }
  },
  "metaData": {
    "status": "draft"
  },
  "errors": []
}
