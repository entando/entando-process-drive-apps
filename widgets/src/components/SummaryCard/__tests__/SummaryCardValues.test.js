import React from 'react';
import { render } from '@testing-library/react';
import SummaryCardValues from 'components/SummaryCard/SummaryCardValues';
import 'mocks/i18nMock';

describe('<SummaryCardValues />', () => {
  it('renders snapshot correctly', () => {
    const values = {
      value: 2123,
      percentage: 0.9243234,
    };
    const { container } = render(<SummaryCardValues dataType="request" values={values} />);

    expect(container).toMatchSnapshot();
  });
});
