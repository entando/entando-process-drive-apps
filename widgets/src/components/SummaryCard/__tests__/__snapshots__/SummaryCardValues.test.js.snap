// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryCardValues /> renders snapshot correctly 1`] = `
<div>
  <div
    class="SummaryCardValues-root-1 upTrend"
  >
    <div>
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        2123
      </h4>
      <span
        class="MuiTypography-root MuiTypography-caption"
      >
        summary.labels.request.total_label
      </span>
    </div>
    <div>
      <span
        class="MuiTypography-root SummaryCardValues-trendValue-3 MuiTypography-body2"
      >
        92%
         
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root SummaryCardValues-valueIcon-4"
          fill="none"
          focusable="false"
          height="11"
          role="presentation"
          viewBox="0 0 8 11"
          width="8"
        >
          <path
            d="M7.38618 4.23054C7.46599 4.07092 7.43697 3.87502 7.32088 3.73716L4.99909 0.951021C4.82496 0.740609 4.46218 0.740609 4.28805 0.951021L1.96626 3.73716C1.85017 3.87502 1.82841 4.07092 1.90096 4.23054C1.98077 4.39742 2.1404 4.499 2.32179 4.499H3.71486V9.14257H1.39307C1.32777 9.14257 1.25522 9.17159 1.21168 9.22238L0.0507891 10.6155C-0.00725558 10.688 -0.0145112 10.7823 0.0217667 10.8694C0.0580446 10.9492 0.145112 11 0.232179 11H5.34011C5.47071 11 5.57229 10.8984 5.57229 10.7678V4.499H6.96536C7.14675 4.499 7.30637 4.39742 7.38618 4.23054Z"
            fill="currentcolor"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
